InvalidPathException when running test cases in embedded mode on Windows

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

When I OrganizationMemberTest locally on Windows, I get: 

```
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#configureKeycloakSessionFactory threw an exception: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /D:/workspace/keycloak/keycloak/tests/custom-scripts/target/classes/META-INF/keycloak-scripts.json
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:204)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:175)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:231)
	at java.base/java.nio.file.Path.of(Path.java:148)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.lambda$getScriptProviderDescriptorsFromClassPath$26(KeycloakProcessor.java:1042)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.configureScriptDescriptor(KeycloakProcessor.java:1104)
```

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

It should work on Windows

### Actual behavior

It does not work on Windows

### How to Reproduce?

Run on Windows

### Anything else?

I'll create a PR.

Full stack trace: 

```
Updating the configuration and installing your custom providers, if any. Please wait.
2026-08-02 09:41:20,572 WARN  [org.keycloak.common.Profile] (build-18) Deprecated features identity-brokering-api:v1, twitter-broker:v1 enabled by default. Check the upgrading guide for steps to use later versions if available.
2026-08-02 09:41:25,183 ERROR [testinfo] (main)     FAILED - org.keycloak.tests.organization.member.OrganizationMemberTest / testDeleteMembersOnOrganizationRemoval

java.lang.RuntimeException: Failed to start the server

	at org.keycloak.Keycloak.start(Keycloak.java:223)
	at org.keycloak.Keycloak$Builder.start(Keycloak.java:140)
	at org.keycloak.testframework.server.EmbeddedKeycloakServer.start(EmbeddedKeycloakServer.java:29)
	at org.keycloak.testframework.server.AbstractKeycloakServerSupplier.getValue(AbstractKeycloakServerSupplier.java:92)
	at org.keycloak.testframework.server.AbstractKeycloakServerSupplier.getValue(AbstractKeycloakServerSupplier.java:24)
	at org.keycloak.testframework.injection.Registry.deployRequestedInstances(Registry.java:254)
	at org.keycloak.testframework.injection.Registry.beforeEach(Registry.java:132)
	at org.keycloak.testframework.KeycloakIntegrationTestExtension.beforeEach(KeycloakIntegrationTestExtension.java:30)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#configureKeycloakSessionFactory threw an exception: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /D:/workspace/keycloak/keycloak/tests/custom-scripts/target/classes/META-INF/keycloak-scripts.json
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:204)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:175)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:231)
	at java.base/java.nio.file.Path.of(Path.java:148)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.lambda$getScriptProviderDescriptorsFromClassPath$26(KeycloakProcessor.java:1042)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.configureScriptDescriptor(KeycloakProcessor.java:1104)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.getScriptProviderDescriptorsFromClassPath(KeycloakProcessor.java:1040)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.loadDeployedScriptProviders(KeycloakProcessor.java:1006)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.loadFactories(KeycloakProcessor.java:962)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.configureKeycloakSessionFactory(KeycloakProcessor.java:673)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:898)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:242)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:441)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:355)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:65)
	at org.keycloak.Keycloak.start(Keycloak.java:216)
	... 9 more
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#configureKeycloakSessionFactory threw an exception: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /D:/workspace/keycloak/keycloak/tests/custom-scripts/target/classes/META-INF/keycloak-scripts.json
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:204)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:175)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:231)
	at java.base/java.nio.file.Path.of(Path.java:148)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.lambda$getScriptProviderDescriptorsFromClassPath$26(KeycloakProcessor.java:1042)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.configureScriptDescriptor(KeycloakProcessor.java:1104)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.getScriptProviderDescriptorsFromClassPath(KeycloakProcessor.java:1040)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.loadDeployedScriptProviders(KeycloakProcessor.java:1006)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.loadFactories(KeycloakProcessor.java:962)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.configureKeycloakSessionFactory(KeycloakProcessor.java:673)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:898)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:242)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:139)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:76)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:180)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:437)
	... 12 more
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /D:/workspace/keycloak/keycloak/tests/custom-scripts/target/classes/META-INF/keycloak-scripts.json
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:204)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:175)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:231)
	at java.base/java.nio.file.Path.of(Path.java:148)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.lambda$getScriptProviderDescriptorsFromClassPath$26(KeycloakProcessor.java:1042)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.configureScriptDescriptor(KeycloakProcessor.java:1104)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.getScriptProviderDescriptorsFromClassPath(KeycloakProcessor.java:1040)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.loadDeployedScriptProviders(KeycloakProcessor.java:1006)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.loadFactories(KeycloakProcessor.java:962)
	at org.keycloak.quarkus.deployment.KeycloakProcessor.configureKeycloakSessionFactory(KeycloakProcessor.java:673)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:898)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:242)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

```

--- a/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
+++ b/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
@@ -20,7 +20,6 @@ package org.keycloak.quarkus.deployment;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
@@ -1039,14 +1038,9 @@ class KeycloakProcessor {
 
             configureScriptDescriptor(descriptor, fileName -> {
                 // descriptor is at META-INF/
-                Path basePath = Path.of(url.getPath()).getParent().getParent();
-
-                String path = basePath.resolve(fileName).toString();
-                if (!path.startsWith(url.getProtocol())) {
-                    path = url.getProtocol() + ":" + path;
-                }
                 try {
-                    return new URI(path).toURL().openStream();
+                    Path basePath = Path.of(url.toURI()).getParent().getParent();
+                    return basePath.resolve(fileName).toUri().toURL().openStream();
                 } catch (IOException | URISyntaxException e) {
                     throw new RuntimeException("Failed to read script file from: " + fileName);
                 }

--- a/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
+++ b/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
@@ -23,7 +23,6 @@ import java.io.InputStream;
 import java.net.URISyntaxException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
-import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
@@ -1039,9 +1038,8 @@ class KeycloakProcessor {
             configureScriptDescriptor(descriptor, fileName -> {
                 // descriptor is at META-INF/
                 try {
-                    Path basePath = Path.of(url.toURI()).getParent().getParent();
-                    return basePath.resolve(fileName).toUri().toURL().openStream();
-                } catch (IOException | URISyntaxException e) {
+                    return new URL(url, "../" + fileName).openStream();
+                } catch (IOException e) {
                     throw new RuntimeException("Failed to read script file from: " + fileName);
                 }
             });

